apps/explorer/lib/explorer/chain/celo/celo_election_rewards.ex

```diff
@@ -171,6 +230,87 @@ defmodule Explorer.Chain.CeloElectionRewards do
     }
   end
 
+  def get_epoch_rewards(account_hash_list, reward_type_list, from, to, page_number, page_size)
+      when from == nil and to == nil,
+      do:
+        get_epoch_rewards(
+          account_hash_list,
+          reward_type_list,
+          17_280,
+          CeloEpochRewards.get_last_epoch_block_number(),
+          page_number,
+          page_size
+        )
+
+  def get_epoch_rewards(account_hash_list, reward_type_list, from, to, page_number, page_size) when from == nil,
+    do: get_epoch_rewards(account_hash_list, reward_type_list, 17_280, to, page_number, page_size)
+
+  def get_epoch_rewards(account_hash_list, reward_type_list, from, to, page_number, page_size) when to == nil,
+    do:
+      get_epoch_rewards(
+        account_hash_list,
+        reward_type_list,
+        from,
+        CeloEpochRewards.get_last_epoch_block_number(),
+        page_number,
+        page_size
+      )
+
+  def get_epoch_rewards(
+        account_hash_list,
+        group_hash_list,
+        from,
+        to,
+        page_number,
+        page_size
+      ) do
+    query = base_api_address_query()
+    total_query = base_sum_and_count_rewards_api_address_query()
+    offset = (page_number - 1) * page_size
+
+    rewards =
+      query
+      |> block_number_query(from, to)
+      |> reward_type_query(["voter"])
+      |> account_hash_query(account_hash_list)
+      |> group_address_hash_query(group_hash_list)
+      |> offset(^offset)
+      |> limit(^page_size)
+      |> Repo.all()
+
+    total =
```

Comment:
is it possible to calculate these values from the rewards above, rather than making another db query?

Reply:
not really, because those results are paginated... at least not in the case were they might be more items than `page_size`, only case where we could avoid making an extra query would be when `page_number == 1` and (pseudocode) `items_returned_from_query < page_size` (or also when `page_number > 1` as we can calculate the total items returned prior to the last page) because then we know certainly that there are no more items to be fetched from the DB, so yeah there are options to avoid that query conditionally - is that something you meant?

Reply:
fair enough, I was hoping we could avoid an extra round trip to the db